arches/app/models/migrations/10552_add_reference_datatype.py

```diff
@@ -0,0 +1,64 @@
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("models", "10541_controlled_list_manager"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="""
+            INSERT INTO d_data_types(
+                datatype,
+                iconclass,
+                modulename,
+                classname,
+                defaultconfig,
+                configcomponent,
+                configname,
+                isgeometric,
+                defaultwidget,
+                issearchable
+            ) VALUES (
+                'reference',
+                'fa fa-list',
+                'datatypes.py',
+                'ReferenceDataType',
+                '{"controlledList": null, "multiValue": false}',
+                'views/components/datatypes/reference',
+                'reference-datatype-config',
+                FALSE,
+                '19e56148-82b8-47eb-b66e-f6243639a1a8',
+                TRUE
+            );
+
+            INSERT INTO widgets(
+                widgetid,
+                name,
+                component,
+                datatype,
+                defaultconfig
+            ) VALUES (
+                '19e56148-82b8-47eb-b66e-f6243639a1a8',
+                'reference-select-widget',
+                'views/components/widgets/reference-select',
+                'reference',
+                '{"placeholder": "Select an option", "i18n_properties": ["placeholder"]}'
+            );
+
+            """,
+            reverse_sql="""
+            DELETE FROM cards_x_nodes_x_widgets
+                WHERE nodeid IN (SELECT nodeid FROM nodes WHERE datatype = 'reference'); 
+            DELETE FROM edges
+                WHERE domainnodeid IN (SELECT nodeid FROM nodes WHERE datatype = 'reference') 
+                OR rangenodeid IN (SELECT nodeid FROM nodes WHERE datatype = 'reference');
+            DELETE FROM nodes WHERE datatype = 'reference';
```

Comment:
If we develop another datatype soon, it might be nice to have this pulled out into a util, and in that case doing it with Django would be easier I imagine, but no worries for now.

Reply:
Added the constraints_x_nodes in 2703531383f810b596a6b92ffde38dbcd33c7b1c

Reply:
Sounds like we're going to use ` RunSQL.noop` here instead.